### PR TITLE
Lesson7

### PR DESCRIPTION
**Pull request by Bulldog2740**

### CODE REVIEW

File: multithread/atomic/AtomicTestApp.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    public static void main(String[] args) {
+        AtomicInteger atomicInt = new AtomicInteger(0);
+        atomicInt.incrementAndGet();
+        ExecutorService executor = Executors.newFixedThreadPool(2);
+
+        IntStream.range(0, 1000)
+                .forEach(i -> executor.submit(atomicInt::incrementAndGet));
+
+        executor.shutdown();
+atomicInt.getAndSet(999999);
```
Comment (Taniann) [inline multithread/atomic/AtomicTestApp.java]: formate your code. add white spaces here

File: multithread/collection/CopyOnWrite.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        list.add("300$");
+        list.add("350$");
+        list.add("400$");
+        new Thread(() -> {
+            for (String str : list) {
+                System.out.println("Thread1:" + str);
+            }
+        }).start();
+        new Thread(() -> {
+    list.add("999 999 999$");
```
Comment (Taniann) [inline multithread/collection/CopyOnWrite.java]: use "ctrl+alt+l" if you're using intelij idea

File: multithread/locks/readwritelock/ReadThread.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    public ReadThread(ReadWriteLock locker) {
+        this.locker = locker;
+    }
+
+    @Override
+    public void run() {
+        locker.readLock().lock();
+        try {
+            Thread.sleep(1000);
+            ReadWriteLockTestApp.information.forEach(System.out::println);
```
Comment (Taniann) [inline multithread/locks/readwritelock/ReadThread.java]: why do you use ReadWriteLockTestApp here? it is a launch class
correct logic of it

File: multithread/locks/readwritelock/WriteThread.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    public WriteThread(ReadWriteLock locker) {
+        this.locker = locker;
+    }
+
+    @Override
+    public void run() {
+        locker.writeLock().lock();
+        try {
+            Thread.sleep(100);
+            ReadWriteLockTestApp.information.add("First");
```
Comment (Taniann) [inline multithread/locks/readwritelock/WriteThread.java]: the same

File: multithread/locks/reentrantlock/CountThread.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        this.locker = locker;
+    }
+
+    @Override
+    public void run() {
+        locker.lock();
+        try {
+            LockTestApp.resource = 0;
+            for (int i = 1; i <= 5; i++) {
+                System.out.println(Thread.currentThread().getName() + " " + LockTestApp.resource);
```
Comment (Taniann) [inline multithread/locks/reentrantlock/CountThread.java]: dont use LockTestApp here, it is a launch class
correct logic of it

File: multithread/queue/BlockingQueues.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+public class BlockingQueues {
+    public static final int COUNT = 5;
+    public static final int CAPACITY = 10;
+
+
+    public static void main(String[] args) {
+        BlockingQueue kgb = new ArrayBlockingQueue(CAPACITY);
+        new Thread(new Producer(kgb,COUNT)).start();
+       new Thread(new Consumer(kgb,COUNT)).start();
```
Comment (Taniann) [inline multithread/queue/BlockingQueues.java]: pay attention on code formatting

File: multithread/queue/Comsumer.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
+package multithread.queue;
+
+import java.util.concurrent.BlockingQueue;
+
+class Consumer implements Runnable{
+    private BlockingQueue blocking;
+    private int count;
```
Comment (Taniann) [inline multithread/queue/Comsumer.java]: add blank line between fields and constructor

File: multithread/synchronizers/cyclicbarrier/CyclicBarrierTestApp.java [hunk 8]
```diff
@@ -0,0 +1,14 @@
+package multithread.synchronizers.cyclicbarrier;
+
+import java.util.concurrent.CyclicBarrier;
+
+public class CyclicBarrierTestApp {
+    public static final CyclicBarrier BARRIER=new CyclicBarrier(2,new FerryBoat());
```
Comment (Taniann) [inline multithread/synchronizers/cyclicbarrier/CyclicBarrierTestApp.java]: code formatting

File: multithread/locks/readwritelock/ReadThread.java (hunk 3 above)
Comment (Bulldog2740) [reply multithread/locks/readwritelock/ReadThread.java]: but how i can take list to write on it?

**Pull request closed without merging**